File processor image cropping

Cyperghost opened the pull request:
Lets you crop images in two ways:

### Min Max

- Allows the user to crop the image to a specified minimum and maximum size
- Minimum and maximum must have the same aspect ratio

### Exact

- Alles sizes must have the same aspect ratio
- If the image already corresponds to the correct dimensions, no dialog is displayed and the image is uploaded directly
- The cropped image will be resized to the largest possible size

Cyperghost committed:
Install cropperjs

Cyperghost committed:
Add `cropper.min.js`

Cyperghost committed:
Add image cropper configuration

Cyperghost committed:
Output the cropper configuration

Cyperghost committed:
Open crop dialog if `cropperConfiguration` set

Cyperghost committed:
Display error message if images are too small

Cyperghost committed:
Add `cropperContainer` around `cropper-canvas`

Cyperghost committed:
Validate the file type

Cyperghost committed:
Check whether the user has canceled the dialog manually

Cyperghost committed:
Check whether the `cropper-selection` in `MinMaxImageCropper` limits to the minimum and maximum.

Cyperghost committed:
Remove sorting the sizes in js, server already does this

Cyperghost committed:
Correct comment

Cyperghost committed:
Update `cropper.min.js` to the current installed version

Cyperghost committed:
Run tsc

Cyperghost committed:
Add since information

Cyperghost committed:
Run `tsc`

Cyperghost committed:
Merge remote-tracking branch 'origin/6.2' into file-upload-image-crop

# Conflicts:
#	package-lock.json
#	package.json
#	wcfsetup/install/lang/en.xml

dtdesign reviewed (commented):
Looks good so far but the final verdict depends on the implementation of the cover photo so that we can validate the MinMax case with a real use case.

Cyperghost committed:
Merge branch '6.2' into file-upload-image-crop

# Conflicts:
#	.github/workflows/javascript.yml
#	wcfsetup/install/lang/de.xml
#	wcfsetup/install/lang/en.xml

Cyperghost committed:
Implement `FileRuntimeCache`, which also loads the thumbnails

Cyperghost committed:
Add `exifreader`

Cyperghost committed:
Add exifreader to `require.config.js`

Cyperghost committed:
Center the cropper selection in the canvas

Cyperghost committed:
Change the construct to private

Cyperghost committed:
Disable zoom
Set default max width and height for the selection

Cyperghost committed:
Set the default selection to the maximum width

Cyperghost committed:
Round the numbers

Cyperghost committed:
Run `tsc`

Cyperghost committed:
Calculate the size of the cropper selection in relation to the window size

Cyperghost committed:
Fixes the problem when the aspect ratio is less than 1, that the selection has an incorrect resolution

Cyperghost committed:
Recalculate cropper height on window resize.
Set height on container(parent) height

Cyperghost committed:
Center `cropper-canvas`.
Fix, if the image on a side has a large dimension that a white area is displayed at the top and/or bottom

Cyperghost committed:
Add helper function to get the inner dimensions for an element

Cyperghost committed:
overwrites default values for `min-width` and `min-height`

Cyperghost committed:
Also note the maximum width that is available

Cyperghost committed:
Merge branch '6.2' into file-upload-image-crop

# Conflicts:
#	.github/workflows/javascript.yml
#	wcfsetup/install/files/js/require.config.js

Cyperghost committed:
Resize the image to the maximum that the browser can display in the dialog.

Cyperghost committed:
Remove round
Limit to max width and height

Cyperghost committed:
Find the exact size for the image or use the minimum size for the image
